docs/architecture-decisions/multiple-sync-sources.md

```diff
@@ -0,0 +1,82 @@
+
+## Background
+
+Flagd daemon is a feature flag evaluation engine that forms a core part of the OpenFeature ecosystem. Unlike OpenFeature SDK components, which are, by design, agnostic to the specifics of flag structure, evaluation, and persistence, Flagd must take an opinionated stance about how feature-flags look, feel, and act. What schema best describes a flag? How should they be evaluated? And in what sort of persistence layer should they be stored?
+
+This latter-most question -- _how should they be stored_ -- is the most opaque design detail of every commercial flagging product from the perspective its end-users. As a front-line engineer using a commercial flagging product, I may, for example, be exposed to flag schema by the product's SDK, and become familiar with its evaluation intricacies over time as my needs grow to require advanced features, or as I encounter surprising behavior. Rarely, however, is an end-user exposed to the details of a commercial product's storage backend. The SaaS vendor is expected to engineer a safe, fast, multi-tenant storage back-end, optimized for its flag schema and operational parameters, and insulate the customer from these details via its SDK. This presents Flagd, an open-source evaluation engine, with an interesting conundrum: what sort of flag storage best suits the needs of its potential user-base (which is everyone)? 
+
+## Requirements
+
+* Support the storage technology that's most likely to meet the needs of current Flagd user-base (Don't be weird. Don't be surprising.)
+* Make it "easy" to extend the flagd runtime to support "new" storage systems
+* Horizontally scalable persistence layer
+* Minimize end-user exposure to persistence "quirks" (replication schemes, leader election, back-end scaling, consistency minutia, etc.. )
+* Reliable, Fast, Transparent
+* Full CRUD, read-optimized.
+
+## Considered Options
+
+* Be super-opinionated and prescribe a built-in raftesque key-value setup, analogous to the designs of k8s and kafka, which prescribe etcd and zookeeper respectively.
+* Roll a single "standard interface" for flag sync (published grpc spec or similar) (??) 
+* Decouple storage from flagd entirely, by exposing a Golang interface type that "providers" can implement to provide support for any data store.
+
+## Proposal
+<!--
+Unsure whether we want a diagram in this section or not. Happy to add one if we want one.
+-->
+The solution to the conundrum posited in the background section of this document is to decouple flag storage entirely from the rest of the runtime, including instead support for myriad commonly used data synchronization interfaces. This allows Flagd to be agnostic to flag storage, while enabling users to use whichever storage back-end best suits their environment.
```

Comment:
What the interface of the storage you have in mind when coming up with this proposal? The existing implementation or there's another new proposal?

Reply:
This is a backfill ADR for the existing ISync interface.